policies/aws_cloudtrail_policies/aws_cloudtrail_enabled.py

```diff
@@ -1,12 +1,25 @@
 def policy(resource):
-    if not resource["Trails"]:
+    # pylint: disable=R1260
+    if not resource.get("Trails"):
         return False
 
-    if not resource["GlobalEventSelectors"]:
+    if not (resource.get("GlobalEventSelectors") or resource.get("GlobalAdvancedEventSelectors")):
         return False
 
-    for selector in resource["GlobalEventSelectors"]:
-        if selector["IncludeManagementEvents"] and selector["ReadWriteType"] == "All":
+    for selector in resource.get("GlobalEventSelectors", [{}]):
+        if selector.get("IncludeManagementEvents") and selector.get("ReadWriteType") == "All":
             return True
 
+    for advanced_selector in resource.get("GlobalAdvancedEventSelectors", [{}]):
+        management_present = False
+        readonly_excluded = True
+        for field_selector in advanced_selector.get("FieldSelectors", [{}]):
+            if field_selector.get("Field") == "eventCategory":
+                event_categories = field_selector.get("Equals", [])
+                if "Management" in event_categories:
+                    management_present = True
+            if field_selector.get("Field") == "readOnly":
+                readonly_excluded = False
+        if readonly_excluded and management_present:
+            return True
```

Comment:
I was comparing these [AWS Docs](https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_AdvancedFieldSelector.html) to this detection. 

It looks to me like these states can exist with these effects
1. readOnly not set as the Field -> both read and write events are logged
2. readOnly set with true in the equals -> only read events are logged
3. readOnly set with false in the equals -> only write events are logged

```suggestion
            if all([
               field_selector.get("Field") == "readOnly",
               "true" in field_selector.get("Equals", [])
               ]):
                readonly_excluded = True
        if management_present and not readonly_excluded:
            return True
```

Reply:
note: I believe this suggestion would cause the test case `Advanced Selector With Management and readOnly` to fail. I think that's what we want. It reads to me that the settings for this test case are logging only read events and excluding write events.

Reply:
Correct, there are three states of readOnly. The only compliant one is if readOnly is not set at all. If it is true or false it is not compliant. My code is currently just looking for the existence of the readOnly flag along with management and will only return as compliant if management is set along with the absence of readOnly